docs/faq.adoc

```diff
@@ -0,0 +1,58 @@
+
+=== How should I organize my credentials?
+
+It's important to understand the access model of credentials in order to organize them most appropriately.
+Credentials are accessed through a combination of scopes and contexts, which are further categorized into types and domains.
+
+The _context_ of a credential is the path in Jenkins where the credentials are accessed.
+Contexts include the root Jenkins context, jobs, users, agents, views, folders, and generally any other item type.
+
+The _scope_ of a credential can be _global, system,_ or _user,_ which determines the visibility of a credential to a context.
+System-scoped credentials are only visible in the root Jenkins context (typically useful for things like agent SSH keys).
+User-scoped credentials are only visible to a context running as that user (such as a user-provided credentials parameter to a build).
+Global-scoped credentials are visible to their associated contexts and any children contexts; this is the default credential scope.
+
+[IMPORTANT]
+====
+Global scoped credentials in the Jenkins Credentials Provider (system credentials) are accessible to any job in the system!
+====
+
+Credentials _stores_ are the underlying storage mechanism for secrets, and credentials _providers_ combine them with the functionality to manage secrets in a context for a certain set of scopes.
+For example, the Jenkins Credentials Provider uses a system-wide credentials store at the root context and supports both system and global scoped credentials.
+Folder plugins provide a Folder Credentials Provider to anchor credentials storage in a folder along with the advanced authorization strategies possible there.
+User-scoped credentials are primarily accessed through the User Credentials Provider which stores secrets with the user profile in Jenkins.
+
+Credentials _types_ include things like username/password, secret text, SSH key, API token, etc.
+Finally, credentials _domains_ provide a way to help organize credentials by domain name, URI, protocol, and other requirements which can make it much easier to identify the appropriate credentials to use from the UI.
+
+=== What plugins do I need to be secure?
+
+First, ensure that you are following the https://www.jenkins.io/doc/book/system-administration/security/[Jenkins security best practices].
+In particular, you'll need to use a non-trivial authorization strategy such as https://plugins.jenkins.io/matrix-auth/[matrix], https://plugins.jenkins.io/role-strategy/[role-based], or https://plugins.jenkins.io/folder-auth/[folder-based]; the trivial authorization strategies would defeat the purpose of securing your secrets!
+Next, you'll need the https://plugins.jenkins.io/authorize-project/[Authorize Project] plugin so that more fine-grained permission controls are available to configure to jobs so that credentials accessible to one job aren't necessarily accessible to every other job.
```

Comment:
`authorize-project` is not necessary for that purpose; you can use folder-scoped credentials.

Reply:
That's true, though folder-stored (there's no folder scope; there's way too many concepts in this plugin) credentials are a little coarser than what `authorize-project` offers. I'll update this to clarify, though, thanks.